oxidation/libparsec_python/src/protocol/block.rs

```diff
@@ -0,0 +1,174 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
+
+use pyo3::import_exception;
+use pyo3::prelude::*;
+use pyo3::types::{PyBytes, PyType};
+
+use crate::ids::{BlockID, RealmID};
+use parsec_api_protocol::authenticated_cmds::{block_create, block_read};
+
+import_exception!(parsec.api.protocol, ProtocolError);
+
+#[pyclass]
+#[derive(PartialEq, Eq, Clone)]
+pub(crate) struct BlockCreateReq(pub block_create::Req);
+
+#[pymethods]
+impl BlockCreateReq {
+    #[new]
+    fn new(block_id: BlockID, realm_id: RealmID, block: Vec<u8>) -> PyResult<Self> {
+        Ok(Self(block_create::Req {
+            block_id: block_id.0,
+            realm_id: realm_id.0,
+            block,
+        }))
+    }
+
+    fn __repr__(&self) -> PyResult<String> {
+        Ok(format!("{:?}>", self.0))
+    }
+
+    fn dumps<'py>(&self, py: Python<'py>) -> PyResult<&'py PyBytes> {
+        Ok(PyBytes::new(
+            py,
+            &self.0.clone().dumps().map_err(ProtocolError::new_err)?,
+        ))
+    }
+}
+
+#[pyclass]
+#[derive(PartialEq, Eq, Clone)]
+pub(crate) struct BlockCreateRep(pub block_create::Rep);
+
+#[pymethods]
+impl BlockCreateRep {
```

Comment:
Same thing for rep. Of course the ok/error creation methods are custom, but as long as they only have a stutus field they can be easily defined in a macro:
```rust
rep_binding!(
  BlockCreateRep,
  block_create::Rep,
);
rep_binding_status_only_constructor!(BlockCreateRep, Ok);
rep_binding_status_only_constructor!(BlockCreateRep, AlreadyExists);
[...]
```

Reply:
Same